internal/semver/semver.go

```diff
@@ -143,8 +142,8 @@ func (v Semver) MajorMinorEqual(other Semver) bool {
 // IsUpgradeTo returns if a version is an upgrade to another version.
 // It checks if the version of v is greater than the version of other and allows a drift of at most one minor version.
 func (v Semver) IsUpgradeTo(other Semver) error {
-	if v.Compare(other) <= 0 {
-		return compatibility.NewInvalidUpgradeError(v.String(), other.String(), errors.New("current version newer than or equal to new version"))
+	if v.Compare(other) < 0 { // allow an upgrade of the same version, because a devbuild might not have a newer version tag
```

Comment:
We should carefully test this change.
One of the reason we considered equal versions to not be an upgrade was to skip upgrading a chart to the same version. I'm not 100% sure on how the Helm Go client behaves, but this might result in restarting all currently running Pods of an installation, and for cert-manager (or other CRD relying charts), require the user to acknowledge a potentially destructive upgrade.
This was previously avoided by simply skipping a chart if the version did not change

Reply:
reverted this change